magellan/tests/test_debugblocker.py

```diff
@@ -0,0 +1,801 @@
+                                               key='_id')
+        self.assertEqual(len(expected_ret_table), len(actual_ret_table))
+
+    def test_debugblocker_14(self):
+        path_ltable = os.sep.join([debugblocker_datasets_path,
+                                   'test_debugblocker_ltable.csv'])
+        path_rtable = os.sep.join([debugblocker_datasets_path,
+                                   'test_debugblocker_rtable.csv'])
+        path_cand = os.sep.join([debugblocker_datasets_path,
+                                   'test_debugblocker_cand.csv'])
+        ltable = read_csv_metadata(path_ltable, key='ID')
+        rtable = read_csv_metadata(path_rtable, key='book_id')
+        cand_set = read_csv_metadata(path_cand, ltable=ltable, rtable=rtable,
+                                      fk_ltable='ltable_ID',
+                                      fk_rtable='rtable_book_id',
+                                      key='_id')
+        attr_corres = [('title', 'book_title'), ('price', 'price'),
+                       ('desc', 'description'), ('genre', 'book_genre'),
+                       ('year', 'pub_year'), ('lang', 'language'),
+                       ('author', 'author'), ('publisher', 'publisher')]
+        output_size = 1
+        ret_dataframe = db.debug_blocker(ltable, rtable, cand_set,
+                                         output_size, attr_corres)
+        expected_columns = ['_id', 'similarity', 'ltable_ID', 'rtable_book_id',
+                            'ltable_title', 'ltable_desc', 'ltable_year',
+                            'ltable_lang', 'ltable_author', 'ltable_publisher',
+                            'rtable_book_title', 'rtable_description',
+                            'rtable_pub_year', 'rtable_language',
+                            'rtable_author', 'rtable_publisher']
+        self.assertEqual(list(ret_dataframe.columns), expected_columns)
+        ret_record = list(ret_dataframe.ix[0])
+        expected_record = [0, 0.33333333333333331, 2, 'B002',
+                           'Thinking in Java',
+                           'learn how to program in Java', 2000, 'ENG',
+                           'Johnnie Doe', np.nan, 'Thinking in C',
+                           'learn programming in C++', '1990', np.nan,
+                           'Jane Doe', 'BCD publisher']
+        self.assertEqual(expected_record[2], ret_record[2])
+        self.assertEqual(expected_record[3], ret_record[3])
+
+    @raises(AssertionError)
+    def test_debugblocker_15(self):
+        A = read_csv_metadata(path_a, key='ID')
+        B = read_csv_metadata(path_b, key='ID')
+        C = read_csv_metadata(path_c, ltable=A, rtable=B,
+                              fk_ltable='ltable_ID',
+                              fk_rtable='rtable_ID',
+                              key='_id')
+        attr_corres = [('ID', 'ID'), ('birth_year', 'birth_year')]
+        db.debug_blocker(A, B, C, 200, attr_corres)
+
+
+def read_record_list(path):
```

Comment:
rename the global functions so that they start with an _ (underscore), e.g., _read_record_list